res/app/device-list/details/device-list-details-directive.js

```diff
@@ -277,6 +279,20 @@ module.exports = function DeviceListDetailsDirective(
         }
       })()
 
+      // check if new column needs a new scope or not
+      // and build accordingly
+      function buildColumn(col, device) {
```

Comment:
I'm kind of worried about performance here, if each row ends up having a scope. We had huge troubles with the device list performance until I rewrote the whole thing with the current code. It may be a little out of date, but there's `stf generate-fake-device` that might help in testing a larger number of devices in the list. However, since they're fake devices, they won't be getting any realtime updates from the server, so it won't be a true measure of performance.